src/main/java/echobot/utilities/Task.java

```diff
@@ -0,0 +1,78 @@
+package echobot.utilities;
+
+import java.io.Serializable;
+
+/**
+ * Class for tasks
+ */
+public class Task implements Serializable {
+
+    /** Name of task */
+    private String name;
+
+    /** Task description, including time */
+    private String description;
+
+    /** Variable to reflect whether a task is finished */
+    private boolean isDone;
+
+    /** Type of task, can be either a todo, deadline, or event */
+    private Type type;
+
+    /**
+     * Creates a new instance of a Task object
+     *
+     * @param name Name of task
+     * @param type Type of task
+     * @param description Task description, which includes by, to, and from
+     */
+    public Task(String name, Type type, String description) {
+        this.name = name;
+        this.isDone = false;
+        this.type = type;
+        this.description = description.equals("") ? description : "\n\t" + description;
+    }
+
+    /**
+     * Marks a task as done
+     *
+     * @return String output after marking as done
+     */
+    public String markAsDone() {
+        this.isDone = true;
+        return "You have marked this task as done\n" + "\t" + this.convertToString();
+    }
+
+    /**
+     * Marks a task as not done
+     *
+     * @return String output after marking as not done
+     */
+    public String markAsNotDone() {
+        this.isDone = false;
+        return "You have marked this task as not done\n" + "\t" + this.convertToString();
+    }
+
+    /**
+     * Converts the task object to a String text for display
+     *
+     * @return the String representation of the task
+     */
+    public String convertToString() {
```

Comment:
Any reason why you did it this way instead of overriding the toString method?

Reply:
Simply put, I only remembered I can overwrite toString() after about a week implementing this, and at that point I didn't want to change the method fearing that the literal string representation of the tasks might be used in the future.

It was needed to extract it into the file in the hard drive, but at this point, I don't see any difference in coding implementation. It's just that the file will become more readable that way.

Which is why I had raised a concern about the file being unreadable in the forums hahaha

I'll be sure to remember this for the tP, thanks